### PR TITLE
getconf.sh: replace use of awk and cat with shell builtins

### PR DESCRIPTION
**Pull request by E5ten**

### CODE REVIEW

File: getconf.sh [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 ifdef() {
 	printf 'static const struct var %s[] = {\n' "$1"
-	awk '{printf("#ifdef %s\n\t{\"%s\",\t%s},\n#endif\n", $2, $1, $2)}'
+	while read -r def func _; do
+		printf '#ifdef %s\n\t{"%s",\t%s},\n#endif\n' "$func" "$def" "$func"
+	done
```
Comment (michaelforney) [inline getconf.sh:7]: The other change is fine, but I don't see the point of this change. In my opinion it doesn't make things simpler.

awk is a POSIX standard tool, so is available on all platforms we support.

**Review by michaelforney — COMMENTED** [review]
Thanks for the pull-request, but could you send this as a patch to hackers@suckless.org? That way others can review the patch as well.

File: getconf.sh (hunk 1 above)
Comment (E5ten) [reply getconf.sh:7]: Sure, but why not allow the script to work in any situation with a functioning shell, which is already needed? I know printf isn't necessarily builtin, but any shell expected to work has access to it.

Comment (michaelforney) [reply getconf.sh:7]: Have you run into a situation where awk isn't available? I think if you tried to run a system without it, you'd hit all sorts of other issues running various scripts.

Comment (michaelforney) [reply getconf.sh:7]: Anyway, feel free to submit both changes to the mailing list (preferably as separate patches). I can apply the cat change, and we can see if anyone else has an opinion about the awk -> printf change. I will apply it if there is some support for it.

**Comment by michaelforney** [comment]
I pushed the cat part of this pull request as 39ae2ae60f0c419fe7c2132d185d19dfa86190e8. In the future, please send patches to the hackers@suckless.org mailing list.

If you still would like to propose the awk removal change, feel free to send the patch again to get some feedback from the rest of the community.

**Pull request closed without merging**